src/WormholeReceiver.sol

```diff
@@ -51,4 +57,10 @@ abstract contract WormholeReceiver is Ownable, WormholeBase {
     }
     _;
   }
+
+  modifier replayProtect(bytes32 deliveryHash) {
+    if (seenDeliveryVaaHashes[deliveryHash]) revert AlreadyProcessed(deliveryHash);
+    seenDeliveryVaaHashes[deliveryHash] = true;
+    _;
+  }
```

Comment:
It felt weird we have to implement this ourselves, I'd expect them to have a base contract or something to inherit from that has this. So I checked the wormhole solidity SDK contract and they do have a `Base` contract in `WormholeRelayerSDK.sol` that implements a lot of these methods. Why aren't we using that? Can we remove our own `WormholeBase` and in contracts do `import {Base as WormholeBase} from "wormhole-solidity-sdk/src/WormholeRelayerSDK.sol"`?

Reply:
The main reason I shied away from using it was that it isn't audited, is actively worked on, and their testing isn't comprehensive for example they don't test the modifiers directly. I thought it would be safer for us to re-implement, but I am not opposed to inheriting their `Base` contract as the functions we are using are very simple.

If we were to use Wormhole's SDK contract we would still need our `WormholeBase` contract because our sender modifier is slightly different then theirs.